lib/phases/phase.js

```diff
@@ -299,9 +299,15 @@ Phase.prototype.fail = function(err) {
       this.resetTimeout();
       this.results = [];
 
+      // clear any outstanding listeners
+      phase.dispatcher.removeAllListeners();
+
       this.device.log
         .clear()
         .then(function() {
+          return phase.capture();
```

Comment:
Since `phase` isn't the one originally calling `capture()`, it probably shouldn't be here either. I know it's redundant, but can you move the `removeAllListeners()` and `capture()` to the individual phases' `retry()` method?